internal/kv/tikv/txn_tikv.go

```diff
@@ -0,0 +1,805 @@
+		return logging_error
+	}
+	CheckElapseAndWarn(start, "Slow txnTiKV MultiRemove() operation", zap.Strings("keys", keys))
+	return nil
+}
+
+// RemoveWithPrefix removes the keys for the given prefix.
+func (kv *txnTiKV) RemoveWithPrefix(prefix string) error {
+	start := time.Now()
+	prefix = path.Join(kv.rootPath, prefix)
+	ctx, cancel := context.WithTimeout(context.Background(), RequestTimeout)
+	defer cancel()
+
+	var logging_error error
+	defer logWarnOnFailure(&logging_error, "txnTiKV RemoveWithPrefix() error", zap.String("prefix", prefix))
+
+	startKey := []byte(prefix)
+	endKey := tikv.PrefixNextKey(startKey)
+	_, err := kv.txn.DeleteRange(ctx, startKey, endKey, 1)
+	if err != nil {
+		logging_error = errors.Wrap(err, "Failed to DeleteRange for RemoveWithPrefix")
+		return logging_error
+	}
+	CheckElapseAndWarn(start, "Slow txnTiKV RemoveWithPrefix() operation", zap.String("prefix", prefix))
+	return nil
+}
+
+// MultiSaveAndRemove saves the key-value pairs and removes the keys in a transaction.
+func (kv *txnTiKV) MultiSaveAndRemove(saves map[string]string, removals []string) error {
+	start := time.Now()
+	ctx, cancel := context.WithTimeout(context.Background(), RequestTimeout)
+	defer cancel()
+
+	var logging_error error
+	defer logWarnOnFailure(&logging_error, "txnTiKV MultiSaveAndRemove error", zap.Any("saves", saves), zap.Strings("removes", removals), zap.Int("saveLength", len(saves)), zap.Int("removeLength", len(removals)))
+
+	txn, err := beginTxn(kv.txn)
+	if err != nil {
+		logging_error = errors.Wrap(err, "Failed to create txn for MultiSaveAndRemove")
+		return logging_error
+	}
+
+	// Defer a rollback only if the transaction hasn't been committed
+	defer rollbackOnFailure(&logging_error, txn)
+
+	for key, value := range saves {
+		key = path.Join(kv.rootPath, key)
+		// Check if value is empty or taking reserved EmptyValue
+		byte_value, err := convertEmptyStringToByte(value)
+		if err != nil {
+			logging_error = errors.Wrap(err, fmt.Sprintf("Failed to cast to byte (%s:%s) for MultiSaveAndRemove", key, value))
+			return logging_error
+		}
+		err = txn.Set([]byte(key), byte_value)
+		if err != nil {
+			logging_error = errors.Wrap(err, fmt.Sprintf("Failed to set (%s:%s) for MultiSaveAndRemove", key, value))
+			return logging_error
+		}
+	}
+
+	for _, key := range removals {
+		key = path.Join(kv.rootPath, key)
+		if err = txn.Delete([]byte(key)); err != nil {
+			logging_error = errors.Wrap(err, fmt.Sprintf("Failed to delete %s for MultiSaveAndRemove", key))
+			return logging_error
+		}
+	}
+
+	err = kv.executeTxn(txn, ctx)
+	if err != nil {
+		logging_error = errors.Wrap(err, "Failed to commit for MultiSaveAndRemove")
+		return logging_error
+	}
+	CheckElapseAndWarn(start, "Slow txnTiKV MultiSaveAndRemove() operation", zap.Any("saves", saves), zap.Strings("removals", removals))
+	return nil
+}
+
+// MultiRemoveWithPrefix removes the keys with given prefix.
+func (kv *txnTiKV) MultiRemoveWithPrefix(prefixes []string) error {
```

Comment:
should we remove the MultiRemvoeWithPrefix support?
I thought this is not used and shouldn't be used since the txn might be very large

Reply:
Hmm, it's not removed from interface as well as etcd 
Let's clean them up in a separate PR (https://github.com/milvus-io/milvus/issues/26830)